sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/QueryPlan.scala

```diff
@@ -180,10 +180,14 @@ abstract class QueryPlan[PlanType <: QueryPlan[PlanType]] extends TreeNode[PlanT
    *             rewrite attribute references in the parent nodes.
    * @param skipCond a boolean condition to indicate if we can skip transforming a plan node to save
    *                 time.
+   * @param canGetOutput a boolean condition to indicate if we can get the output of a plan node
+   *                     to prune the attributes mapping to be propagated. The default value is true
+   *                     as only unresolved logical plan can't get output.
```

Comment:
nit: Is this needed to be parameterized? Seems we only need check unresolved logical plan.

Reply:
Do you mean something like
```
val canGetOutput = plan match {
  case l: LogicalPlan => l.resolved
  case _ => true
}
```
then we can remove `skipCond` as well by matching `AnalysisHelper`.

Reply:
One advantage of the current way is we can skip the check for optimizer rules (use the default value true).

Reply:
Oh, I see. Make sense.